internal/signaling/peer.go

```diff
@@ -250,6 +292,15 @@ func (p *Peer) HandleClosePacket(ctx context.Context, packet ClosePacket) error
 				logger.Error("failed to publish disconnect packet", zap.Error(err))
 			}
 		}
+
+		result, err := p.store.DoLeaderElection(ctx, p.Game, p.Lobby)
+		if err != nil {
+			logger.Error("failed to do leader election", zap.Error(err))
+		} else if result != nil {
+			if err := p.publishLeaderToLobby(ctx, result); err != nil {
+				return err
+			}
+		}
```

Comment:
Could store.DoLeaderElection be part of p.publishLeaderToLobby? You call it everywhere in the same order.

Reply:
Not everywhere. And that would tightly couple pubsub to the store which we didn't want. Pubsub happens to go though the store currently, but that might not always be the case.

Reply:
No, I meant in p.publishLeaderToLobby you also call the p.store.DoLeaderElection (not the otherway around).

Reply:
Ok, that is indeed slightly cleaner, pushed it now.